File: test/data/src/py/VernalizationProgress.py

if (isVernalizable==1 and vernaprog < 1):
    tt = deltaTT       
    if (tt >= minTvern and tt <= intTvern):
        vernaprog = vernaprog + vAI * tt + vBEE      
    if (tt > intTvern):
        maxVernaProg = vAI * intTvern + vBEE
        dLverna = max(minDL, min(maxDL, dayLength))
        vernaprog += max(0, maxVernaProg * (1 + ((intTvern - tt) / (maxTvern - intTvern)) * ((dLverna - minDL) / (maxDL - minDL))))      
    primordno = 2.0 * leafNumber + pNini     
    minLeafNumber = minFinalNumber
    if (vernaprog >= 1.0 or primordno >= aMXLFNO):      
        minFinalNumber = max(primordno, minFinalNumber)
        calendarMoments.append("EndVernalisation")
        calendarCumuls.append(cumulTT) 
        calendarDates.append(currentdate)        
        vernaprog = max(1, vernaprog)      
    else:    
        potlfno = aMXLFNO - (aMXLFNO - minLeafNumber) * vernaprog
        if (primordno >= potlfno):    
            minFinalNumber = max((potlfno + primordno) / 2.0, minFinalNumber)
            vernaprog = max(1, vernaprog)
            calendarMoments.append("EndVernalisation")
            calendarCumuls.append(cumulTT) 
            calendarDates.append(currentdate)

             

File: test/data/phenology/crop2ml/algo/f90/VernalizationProgress.f90

    !use crop2mlModules
        REAL:: tt, maxVernaProg, dLverna, primordno, minLeafNumber, potlfno
        IF ((isVernalizable==1) .AND. (vernaprog < 1)) THEN
            tt = deltaTT
            IF ((tt >= minTvern) .AND. (TT <= intTvern)) THEN
                vernaprog = vernaprog + vAI * tt + vBEE
            END IF
            IF (tt > intTvern) THEN
                maxVernaProg = vAI * intTvern + vBEE
                dLverna = MAX(minDL, MIN(maxDL, dayLength))
                vernaprog =vernaprog + MAX(0.0, maxVernaProg * (1 + ((intTvern - tt) / (maxTvern - intTvern))&
                    * ((dLverna - minDL) / (maxDL - minDL))))
            END IF
            primordno = 2.0 * leafNumber + pNini
            minLeafNumber = minFinalNumber
            IF ((vernaprog >= 1.0) .OR. (primordno >= aMXLFNO)) THEN
                minFinalNumber = MAX(primordno, minFinalNumber)
                CALL AddToListChar(calendarMoments, "EndVernalisation")
                CALL AddToList(calendarCumuls, cumulTT)
                CALL AddToListChar(calendarDates, currentdate)
            ELSE
                potlfno = aMXLFNO - (aMXLFNO - minLeafNumber) * vernaprog
                IF (primordno >= potlfno) THEN
                    minFinalNumber = MAX((potlfno + primordno) / 2.0, minFinalNumber)
                    CALL AddToListChar(calendarMoments,"EndVernalisation")
                    CALL AddToList(calendarCumuls,cumulTT)
                    CALL AddToListChar(calendarDates,currentdate)
                    vernaprog = MAX(1, INT(vernaprog))
                END IF
            END IF
        END IF





